src/scripts/main.js

```diff
@@ -1,3 +1,48 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const lines = table.getElementsByTagName('tr');
+const tableBody = table.querySelector('tbody');
+const counter = lines[0].getElementsByTagName('td');
```

Comment:
The variable `counter` is initialized to count the number of columns in the first row. This might not accurately reflect the current state of the table if columns are added or removed dynamically. Consider recalculating the number of columns dynamically within the event listeners.